packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -779,32 +787,107 @@ export class AbstractQueryGeneratorTypeScript {
   }
 
   bulkDeleteQuery(tableName: TableNameOrModel, options: BulkDeleteQueryOptions): string {
-    const table = this.quoteTable(tableName);
-    const whereOptions = isModelStatic(tableName) ? { ...options, model: tableName } : options;
+    return joinSQLFragments([
+      `DELETE FROM ${this.quoteTable(tableName)}`,
+      this.#internals.addWhereSelectLimitOffset(tableName, options),
+    ]);
+  }
 
-    if (options.limit && this.dialect.supports.delete.modelWithLimit) {
-      if (!isModelStatic(tableName)) {
-        throw new Error('Cannot use LIMIT with bulkDeleteQuery without a model.');
+  /**
+   * Returns an update query
+   *
+   * @param tableName
+   * @param attrValueHash
+   * @param options
+   * @param columnDefinitions
+   */
+  updateQuery(
+    tableName: TableNameOrModel,
```

Comment:
In order to support the in progress repository mode, you should use `TableOrModel` instead of `TableNameOrModel` now

The difference is that `TableOrModel` also accepts a `modelDefinition`

You can then use `extractModelDefinition` or `extractTableIdentifier` utils on it to access the `ModelDefinition` or `TableName` respectively

Requires merging main into this branch

Reply:
Done